tensorflow_probability/python/optimizer/bfgs.py

```diff
@@ -276,10 +277,9 @@ def _body(state):
           state, inverse_hessian_estimate=actual_inv_hessian)
 
       next_state = bfgs_utils.line_search_step(
-          current_state,
-          value_and_gradients_function, actual_search_direction,
+          current_state, value_and_gradients_function, actual_search_direction,
           tolerance, f_relative_tolerance, x_tolerance, stopping_condition,
-          max_line_search_iterations)
+          max_line_search_iterations, **line_search_kwargs)
```

Comment:
The purpose is for the user to be able to control the parameters of the actual line search algorithm, namely `linesearch.hager_zhang`.  All of the parameters that `line_search_step` itself consumes are already being passed in explicitly.  Please pass the dictionary through to that depth, and add a unit test that you can, indeed, affect the behavior of the algorithm by controlling at least one of those parameters (e.g., `initial_step_size`).